rest-service/manager_rest/storage/storage_manager.py

```diff
@@ -629,3 +673,95 @@ def __iter__(self):
 
     def __getitem__(self, item):
         return self.items[item]
+
+
+class SQLAlchemyTracingHelper(object):
+    @staticmethod
+    def enable_tracing_for_all_engines():
+        listen(Engine, 'before_cursor_execute',
+               SQLAlchemyTracingHelper._engine_before_cursor_handler)
+        listen(Engine, 'after_cursor_execute',
+               SQLAlchemyTracingHelper._engine_after_cursor_handler)
+        listen(Engine, 'handle_error',
+               SQLAlchemyTracingHelper._engine_error_handler)
+
+    @staticmethod
+    def _clear_traced(obj):
+        '''
+        Clear an object's decorated tracing fields,
+        to prevent unintended further tracing.
+        '''
+        if hasattr(obj, '_parent_span'):
+            del obj._parent_span
+        if hasattr(obj, '_traced'):
+            del obj._traced
+
+    @staticmethod
+    def _normalize_stmt(statement):
+        return statement.strip().replace('\n', '').replace('\t', '')
+
+    @staticmethod
+    def _engine_before_cursor_handler(conn, cursor,
+                                      statement, parameters,
+                                      context, executemany):
+        current_span = get_current_span()
+        if not current_span:
+            return
+
+        stmt_obj = None
+        if context.compiled is not None:
+            stmt_obj = context.compiled.statement
```

Comment:
Those three lines can be reduced to a single line

Reply:
Done.